include/robot_interfaces/robot_backend.hpp

```diff
@@ -26,12 +26,113 @@
 
 namespace robot_interfaces
 {
+/**
+ * @brief Timer to measure code execution time with "checkpoints"
+ *
+ * This timer is meant to be used for measuring execution time of a loop.  It
+ * measures time between calls of the `start` method, so by calling this at the
+ * beginning of the loop, you get the execution time of the full iteration.
+ * Further, you can define "checkpoints" within the loop to measure time of
+ * separate steps in the loop.  Call the `checkpoint` method after the code that
+ * is associated with it.  For each checkpoint, the time elapsed since the last
+ * checkpoint is measured (`start` counts as a checkpoint in this regard).
+ *
+ * Example:
+ *
+ * ~~~{.cpp}
+ * CheckpointTimer<3> timer;
+ *
+ * while (condition)
+ * {
+ *     timer.start();
+ *
+ *     initialize()
+ *     timer.checkpoint("initialize");
+ *
+ *     do();
+ *     some();
+ *     stuff();
+ *     timer.checkpoint("do some stuff");
+ *
+ *     write_log();
+ *     timer.checkpoint("logging");
+ * }
+ * ~~~
+ *
+ * @tparam NUM_CHECKPOINTS Number of checkpoints.
+ * @tparam ENABLED Set to false, to disable timer.  Method calls will have no
+ * effect (and should hopefully be optimized away by the compiler).
+ */
+template <size_t NUM_CHECKPOINTS, bool ENABLED = true>
+class CheckpointTimer
```

Comment:
i think this class is very useful! maybe it would make sense to integrate it into the real_time_tools::Timer class?

Reply:
Not sure if integrating it into the `Timer` class makes sense (it uses this class internally) but I am going to move it to the `real_time_tools` package.